src/reconcile/rec_visibility.c

```diff
@@ -49,48 +49,34 @@ __rec_update_save(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, voi
  *     Append the key's original value to its update list.
  */
 static int
-__rec_append_orig_value(WT_SESSION_IMPL *session, WT_PAGE *page, WT_INSERT *ins, void *ripcip,
-  WT_UPDATE *upd, WT_CELL_UNPACK *unpack)
+__rec_append_orig_value(
+  WT_SESSION_IMPL *session, WT_PAGE *page, WT_UPDATE *upd, WT_CELL_UNPACK *unpack)
 {
     WT_DECL_ITEM(tmp);
     WT_DECL_RET;
-    WT_PAGE_MODIFY *mod;
-    WT_UPDATE *append, **upd_entry;
+    WT_UPDATE *append, *tombstone;
     size_t size;
 
-    mod = page->modify;
-    upd_entry = NULL;
+    WT_ASSERT(session, upd != NULL);
+    /* Done if at least one self-contained update is globally visible. */
```

Comment:
There's no need for the `WT_ASSERT` -- we're about to use `upd`, and we'll drop core if it's NULL.

Reply:
Thanks, removed.